lib/plugins.js

```diff
@@ -67,11 +67,11 @@ class PluginContext {
   registerCondition (condition) {
     this.conditions.push(condition);
   }
-  registerGatewayExtension (gatewayExtension) {
-    this.gatewayExtensions.push(gatewayExtension);
+  registerGatewayRoute (gatewayRoute) {
```

Comment:
Should this be plural?  We can register more than one route at a time through this, right?  @altsang @DrMegavolt

Reply:
@kevinswiber this should be singular because it accepts only one non array parameter
but the naming of argument could be 
gatewayRoutesDeclaration